controllers/openstackmachine_controller.go

```diff
@@ -280,11 +289,133 @@ func (r *OpenStackMachineReconciler) reconcileDelete(scope scope.Scope, cluster
 		return ctrl.Result{}, fmt.Errorf("delete instance: %w", err)
 	}
 
+	if err := r.reconcileDeleteFloatingAddressFromPool(scope, openStackMachine); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	controllerutil.RemoveFinalizer(openStackMachine, infrav1.MachineFinalizer)
 	scope.Logger().Info("Reconciled Machine delete successfully")
 	return ctrl.Result{}, nil
 }
 
+// reconcileFloatingAddressFromPool creates IPAddressClaims for the OpenStackMachine if the OpenStackMachine has a
+// FloatingAddressFromPool specified. It then associates the IPAddressClaim with the instance.
+func (r *OpenStackMachineReconciler) reconcileFloatingAddressFromPool(ctx context.Context, scope scope.Scope, openStackMachine *infrav1.OpenStackMachine, openStackCluster *infrav1.OpenStackCluster, instanceStatus *compute.InstanceStatus, instanceNS *compute.InstanceNetworkStatus) error {
+	if openStackMachine.Spec.FloatingAddressFromPool == nil {
+		conditions.MarkTrue(openStackMachine, infrav1.FloatingAddressFromPoolReadyCondition)
+		openStackMachine.Status.FloatingAddressFromPoolReady = true
```

Comment:
seems this means if there is no need to assign address, the status still tells floating address is ready
not sure this may make some confusion?

Reply:
This was probably more reasonable when I had a list of pools instead of a single pool, I'll remove this part and just add the nil check to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1762/files#diff-e0a0984c0e26655878e2f94623fbf8cddb4d8a89e3443c1296dfaadc8802838aR499